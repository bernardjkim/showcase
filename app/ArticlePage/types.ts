import { Article } from 'types';
export interface Comment {}
export interface Like {}
export interface Query {
  id: string;
}
export interface ArticleState {
  loading: boolean;
  error?: Error;
  article?: Article;
  comments: Comment[];
  likes: Like[];
}
// =============================================================================
//  ACTIONS
// =============================================================================

export const CREATE_COMMENT = 'app/ArticlePage/CREATE_COMMENT';
export const CREATE_COMMENT_ERROR = 'app/ArticlePage/CREATE_COMMENT_ERROR';
export const CREATE_COMMENT_SUCCESS = 'app/ArticlePage/CREATE_COMMENT_SUCCESS';

export const LIKE_ARTICLE = 'app/ArticlePage/LIKE_ARTICLE';
export const LIKE_ARTICLE_ERROR = 'app/ArticlePage/LIKE_ARTICLE_ERROR';
export const LIKE_ARTICLE_SUCCESS = 'app/ArticlePage/LIKE_ARTICLE_SUCCESS';

export const LOAD_ARTICLE = 'app/ArticlePage/LOAD_ARTICLE';
export const LOAD_ARTICLE_ERROR = 'app/ArticlePage/LOAD_ARTICLE_ERROR';
export const LOAD_ARTICLE_SUCCESS = 'app/ArticlePage/LOAD_ARTICLE_SUCCESS';

export const LOAD_COMMENTS = 'app/ArticlePage/LOAD_COMMENTS';
export const LOAD_COMMENTS_ERROR = 'app/ArticlePage/LOAD_COMMENTS_ERROR';
export const LOAD_COMMENTS_SUCCESS = 'app/ArticlePage/LOAD_COMMENTS_SUCCESS';

export const LOAD_LIKES = 'app/ArticlePage/LOAD_LIKES';
export const LOAD_LIKES_ERROR = 'app/ArticlePage/LOAD_LIKES_ERROR';
export const LOAD_LIKES_SUCCESS = 'app/ArticlePage/LOAD_LIKES_SUCCESS';

export interface CreateCommentAction {
  type: typeof CREATE_COMMENT;
  comment: Comment;
}

export interface CreateCommentErrorAction {
  type: typeof CREATE_COMMENT_ERROR;
  error: Error;
}

export interface CreateCommentSuccessAction {
  type: typeof CREATE_COMMENT_SUCCESS;
  comment: Comment;
}

export interface LikeArticleAction {
  type: typeof LIKE_ARTICLE;
}

export interface LikeArticleErrorAction {
  type: typeof LIKE_ARTICLE_ERROR;
  error: Error;
}

export interface LikeArticleSuccessAction {
  type: typeof LIKE_ARTICLE_SUCCESS;
  like: Like;
}

export interface LoadArticleAction {
  type: typeof LOAD_ARTICLE;
  query: Query;
}

export interface LoadArticleErrorAction {
  type: typeof LOAD_ARTICLE_ERROR;
  error: Error;
}

export interface LoadArticleSuccessAction {
  type: typeof LOAD_ARTICLE_SUCCESS;
  article: Article;
}

export interface LoadCommentsAction {
  type: typeof LOAD_COMMENTS;
}

export interface LoadCommentsErrorAction {
  type: typeof LOAD_COMMENTS_ERROR;
  error: Error;
}

export interface LoadCommentsSuccessAction {
  type: typeof LOAD_COMMENTS_SUCCESS;
  comments: Comment[];
}

export interface LoadLikesAction {
  type: typeof LOAD_LIKES;
}

export interface LoadLikesErrorAction {
  type: typeof LOAD_LIKES_ERROR;
  error: Error;
}

export interface LoadLikesSuccessAction {
  type: typeof LOAD_LIKES_SUCCESS;
  likes: Like[];
}

export type ArticleActionTypes =
  | CreateCommentAction
  | CreateCommentErrorAction
  | CreateCommentSuccessAction
  | LikeArticleAction
  | LikeArticleErrorAction
  | LikeArticleSuccessAction
  | LoadArticleAction
  | LoadArticleErrorAction
  | LoadArticleSuccessAction
  | LoadCommentsAction
  | LoadCommentsErrorAction
  | LoadCommentsSuccessAction
  | LoadLikesAction
  | LoadLikesErrorAction
  | LoadLikesSuccessAction;
