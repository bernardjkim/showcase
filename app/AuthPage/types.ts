export interface SignupFormData {
  username: string;
  email: string;
  password: string;
  passwordConfirm: string;
}

export interface LoginFormData {
  email: string;
  password: string;
}

export interface AuthState {
  loading: boolean;
  // TODO: Error Def
  error: any;
}

export const CLEAR_ERRORS = 'app/AuthPage/CLEAR_ERRORS';

export const CREATE_USER = 'app/AuthPage/CREATE_USER';
export const CREATE_USER_SUCCESS = 'app/AuthPage/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'app/AuthPage/CREATE_USER_ERROR';

export const CREATE_TOKEN = 'app/App/CREATE_TOKEN';
export const CREATE_TOKEN_SUCCESS = 'app/App/CREATE_TOKEN_SUCCESS';
export const CREATE_TOKEN_ERROR = 'app/App/CREATE_TOKEN_ERROR';

export interface ClearErrorsAction {
  type: typeof CLEAR_ERRORS;
}

export interface CreateUserAction {
  type: typeof CREATE_USER;
  form: SignupFormData;
}

export interface CreateUserErrorAction {
  type: typeof CREATE_USER_ERROR;
  error: Error;
}

export interface CreateUserSuccessAction {
  type: typeof CREATE_USER_SUCCESS;
}

export interface CreateTokenAction {
  type: typeof CREATE_TOKEN;
  form: LoginFormData;
}

export interface CreateTokenErrorAction {
  type: typeof CREATE_TOKEN_ERROR;
  error: Error;
}

export interface CreateTokenSuccessAction {
  type: typeof CREATE_TOKEN_SUCCESS;
}

export type AuthActionTypes =
  | ClearErrorsAction
  | CreateTokenAction
  | CreateTokenErrorAction
  | CreateTokenSuccessAction
  | CreateUserAction
  | CreateUserErrorAction
  | CreateUserSuccessAction;
