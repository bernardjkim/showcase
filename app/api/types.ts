import { Sort } from 'types';

export interface ArticleApi {
  create: string;
  getAll: (offset: number) => string;
  getOne: (id: string) => string;
  search: (term: string, offset: number, sort: Sort) => string;
}

export interface AuthApi {
  login: string;
  logout: string;
}

export interface CommentApi {
  create: string;
  get: (id: string) => string;
}

export interface LikeApi {
  getAll: string;
  create: string;
  getByArticle: (id: string) => string;
}

export interface UserApi {
  create: string;
  getCurrent: string;
  getOne: (id: string) => string;
}

export interface API {
  article: ArticleApi;
  auth: AuthApi;
  comment: CommentApi;
  like: LikeApi;
  user: UserApi;
}
