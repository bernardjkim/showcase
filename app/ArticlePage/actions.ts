/*
 *
 * ArticlePage actions
 *
 */

import { Article } from 'types';
import {
  ArticleActionTypes,
  Comment,
  CREATE_COMMENT,
  CREATE_COMMENT_ERROR,
  CREATE_COMMENT_SUCCESS,
  Like,
  LIKE_ARTICLE,
  LIKE_ARTICLE_ERROR,
  LIKE_ARTICLE_SUCCESS,
  LOAD_ARTICLE,
  LOAD_ARTICLE_ERROR,
  LOAD_ARTICLE_SUCCESS,
  LOAD_COMMENTS,
  LOAD_COMMENTS_ERROR,
  LOAD_COMMENTS_SUCCESS,
  LOAD_LIKES,
  LOAD_LIKES_ERROR,
  LOAD_LIKES_SUCCESS,
  Query,
} from './types';

/**
 * Create comment, this action starts the request saga
 *
 * @param  {object}  comment  - The comment value
 *
 * @return {object}           - An action object with a type of CREATE_COMMENT
 */
export function createComment(comment: Comment): ArticleActionTypes {
  return {
    type: CREATE_COMMENT,
    comment,
  };
}

/**
 * Dispatched when posting the comment is acknowledged by the request saga
 *
 * @param  {object}  comment  - The comment data
 *
 * @return {object}           - An action object with a type of CREATE_COMMENT_SUCCESS
 */
export function createCommentSuccess(comment: Comment): ArticleActionTypes {
  return {
    type: CREATE_COMMENT_SUCCESS,
    comment,
  };
}

/**
 * Dispatched when posting the comment fails
 *
 * @return {object} - An action object with a type of CREATE_COMMENT_SUCCESS
 */
export function createCommentError(error: Error): ArticleActionTypes {
  return {
    type: CREATE_COMMENT_ERROR,
    error,
  };
}

/**
 * Like article, this action starts the request saga
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE
 */
export function likeArticle(): ArticleActionTypes {
  return {
    type: LIKE_ARTICLE,
  };
}

/**
 * Dispatched when posting the like is acknowledged by the request saga
 * @param   {Like}    like  - Like object
 *
 * @returns {object}        - An action object with a type of LIKE_ARTICLE_SUCCESS
 */
export function likeArticleSuccess(like: Like): ArticleActionTypes {
  return {
    type: LIKE_ARTICLE_SUCCESS,
    like,
  };
}

/**
 * Dispatched when posting the like fails
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE_ERROR
 */
export function likeArticleError(error: Error): ArticleActionTypes {
  return {
    type: LIKE_ARTICLE_ERROR,
    error,
  };
}

/**
 * Load article, this action starts the request saga
 *
 * @param  {object}  query  - The comment data
 *
 * @return {object}         - An action object with a type of LOAD_ARTICLE
 */
export function loadArticle(query: Query): ArticleActionTypes {
  return {
    type: LOAD_ARTICLE,
    query,
  };
}

/**
 * Dispatched when the article is loaded by the request saga
 *
 * @param  {object}  article  - The article data
 *
 * @return {object}           - An action object with a type of LOAD_ARTICLE_SUCCESS
 */
export function loadArticleSuccess(article: Article): ArticleActionTypes {
  return {
    type: LOAD_ARTICLE_SUCCESS,
    article,
  };
}

/**
 * Dispatched when loading the article fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_ARTICLE_ERROR
 */
export function loadArticleError(error: Error): ArticleActionTypes {
  return {
    type: LOAD_ARTICLE_ERROR,
    error,
  };
}

/**
 * Load comments, this action starts the request saga
 *
 * @return {object} - An action object with a type of LOAD_COMMENTS
 */
export function loadComments(): ArticleActionTypes {
  return {
    type: LOAD_COMMENTS,
  };
}

/**
 * Dispatched when the comments are loaded by the request saga
 *
 * @param  {array}  comments  - The comments data
 *
 * @return {object}           - An action object with a type of LOAD_COMMENT_SUCCESS
 */
export function loadCommentsSuccess(comments: Comment[]): ArticleActionTypes {
  return {
    type: LOAD_COMMENTS_SUCCESS,
    comments,
  };
}

/**
 * Dispatched when loading the comments fails
 *
 * @param  {object} error - The error
 *
 * @return {object}       - An action object with a type of LOAD_COMMENTS_ERROR
 */
export function loadCommentsError(error: Error): ArticleActionTypes {
  return {
    type: LOAD_COMMENTS_ERROR,
    error,
  };
}

/**
 * Load likes, this action starts the request saga
 *
 * @return {object} - An action object with a type of LOAD_LIKES
 */
export function loadLikes(): ArticleActionTypes {
  return {
    type: LOAD_LIKES,
  };
}

/**
 * Dispatched when the likes are loaded by the request saga
 *
 * @param  {Like[]}  likes  - The array of likes for this article
 *
 * @return {object}         - An action object with a type of LOAD_LIKES_SUCCESS
 */
export function loadLikesSuccess(likes: Like[]): ArticleActionTypes {
  return {
    type: LOAD_LIKES_SUCCESS,
    likes,
  };
}

/**
 * Dispatched when loading the likes fails
 *
 * @param  {object} error - The error
 *
 * @return {object}       - An action object with a type of LOAD_LIKES_ERROR
 */
export function loadLikesError(error: Error): ArticleActionTypes {
  return {
    type: LOAD_LIKES_ERROR,
    error,
  };
}
