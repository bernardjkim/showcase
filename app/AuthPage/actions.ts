/*
 *
 * AuthPage actions
 *
 */

import {
  LoginFormData,
  SignupFormData,
  AuthActionTypes,
  CLEAR_ERRORS,
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './types';

/**
 * Clear error messages
 *
 * @return  {object}  - An action object with a type of CLEAR_ERRORS
 */
export function clearErrors(): AuthActionTypes {
  return {
    type: CLEAR_ERRORS,
  };
}

/**
 * Create token, this action starts the request saga
 *
 * @param  {string}  email    - The user email
 * @param  {string}  password - The user password
 *
 * @return {object}           - An action object with a type of CREATE_TOKEN
 */
export function createToken(form: LoginFormData): AuthActionTypes {
  return {
    type: CREATE_TOKEN,
    form,
  };
}

/**
 * Dispatched when creating the token is successful
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_SUCCESSFUL
 */

export function createTokenSuccess(): AuthActionTypes {
  return {
    type: CREATE_TOKEN_SUCCESS,
  };
}

/**
 * Dispatched when creating the token failed
 *
 * @param   {string}  error - Received error
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_ERROR
 */
export function createTokenError(error: Error): AuthActionTypes {
  return {
    type: CREATE_TOKEN_ERROR,
    error,
  };
}
export function createUser(form: SignupFormData): AuthActionTypes {
  return {
    type: CREATE_USER,
    form,
  };
}

export function createUserSuccess(): AuthActionTypes {
  return {
    type: CREATE_USER_SUCCESS,
  };
}

export function createUserError(error: Error): AuthActionTypes {
  return {
    type: CREATE_USER_ERROR,
    error,
  };
}
