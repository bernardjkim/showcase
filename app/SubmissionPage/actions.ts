/*
 *
 * SubmissionPage actions
 *
 */

import {
  CLEAR_STATE,
  Error,
  Form,
  SubmissionActionTypes,
  SUBMIT_FORM,
  SUBMIT_FORM_ERROR,
  SUBMIT_FORM_SUCCESS,
} from './types';

export function clearState(): SubmissionActionTypes {
  return {
    type: CLEAR_STATE,
  };
}

export function submitForm(form: Form): SubmissionActionTypes {
  return {
    type: SUBMIT_FORM,
    form,
  };
}

export function submitFormSuccess(): SubmissionActionTypes {
  return {
    type: SUBMIT_FORM_SUCCESS,
  };
}

export function submitFormError(error: Error): SubmissionActionTypes {
  return {
    type: SUBMIT_FORM_ERROR,
    error,
  };
}
