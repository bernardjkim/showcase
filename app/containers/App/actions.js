/*
 * 
 * App actions
 * 
 */

import {
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  DELETE_TOKEN,
  DELETE_TOKEN_ERROR,
  DELETE_TOKEN_SUCCESS,
  LOAD_USER,
  LOAD_USER_ERROR,
  LOAD_USER_SUCCESS,
} from './constants';

/**
 * Create token, this action starts the request saga
 *
 * @param  {string}  email    - The user email
 * @param  {string}  password - The user password
 *
 * @return {object}           - An action object with a type of CREATE_TOKEN
 */
export function createToken(email, password) {
  return {
    type: CREATE_TOKEN,
    email,
    password,
  };
}

/**
 * Dispatched when creating the token is successful
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_SUCCESSFUL
 */
export function createTokenSuccess() {
  return {
    type: CREATE_TOKEN_SUCCESS,
  };
}

/**
 * Dispatched when creating the token failed
 *
 * @param   {string}  error - Received error
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_ERROR
 */
export function createTokenError(error) {
  return {
    type: CREATE_TOKEN_ERROR,
    error,
  };
}

/**
 * Delete token
 *
 * @return  {object}  - An action object with a type of DELETE_TOKEN
 */
export function deleteToken() {
  return {
    type: DELETE_TOKEN,
  };
}

/**
 * Dispatched when deleteing token fails
 *
 * @return  {object}  - An action object with a type of DELETE_TOKEN_SUCCESS
 */
export function deleteTokenSuccess() {
  return {
    type: DELETE_TOKEN_SUCCESS,
  };
}

/**
 * Dispatched when deleteing token is successful
 *
 * @param   {Error} error - Error received
 *
 * @return  {object}      - An action object with a type of DELETE_TOKEN_SUCCESS
 */
export function deleteTokenError(error) {
  return {
    type: DELETE_TOKEN_ERROR,
    error,
  };
}

/**
 * Load user, this action start the request saga
 *
 * @return  {object}  - An action object with a type of LOAD_USER
 */
export function loadUser() {
  return {
    type: LOAD_USER,
  };
}

/**
 * Dispatched when loading user is successful
 *
 * @param   {User}  user  - Received user
 *
 * @return  {object}      - An action object with a type of LOAD_USER_SUCCESS
 */
export function loadUserSuccess(user) {
  return {
    type: LOAD_USER_SUCCESS,
    user,
  };
}

/**
 * Dispatched when loading user fails
 *
 * @param   {Error} error - Error received
 *
 * @return  {object}      - An action object with a type of LOAD_USER_ERROR
 */
export function loadUserError(error) {
  return {
    type: LOAD_USER_ERROR,
    error,
  };
}
