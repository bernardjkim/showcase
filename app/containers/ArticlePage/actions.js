/*
 *
 * ArticlePage actions
 *
 */

import {
  CREATE_COMMENT,
  CREATE_COMMENT_ERROR,
  CREATE_COMMENT_SUCCESS,
  LIKE_ARTICLE,
  LIKE_ARTICLE_ERROR,
  LIKE_ARTICLE_SUCCESS,
  LOAD_ARTICLE,
  LOAD_ARTICLE_ERROR,
  LOAD_ARTICLE_SUCCESS,
  LOAD_COMMENTS,
  LOAD_COMMENTS_ERROR,
  LOAD_COMMENTS_SUCCESS,
} from './constants';

/**
 * Create comment, this action starts the request saga
 *
 * @param  {object}  comment  - The comment value
 *
 * @return {object}           - An action object with a type of CREATE_COMMENT
 */
export function createComment(comment) {
  return {
    type: CREATE_COMMENT,
    comment,
  };
}

/**
 * Dispatched when posting the comment is acknowledged by the request saga
 *
 * @param  {object}  comment  - The comment data
 *
 * @return {object}           - An action object with a type of CREATE_COMMENT_SUCCESS
 */
export function createCommentSuccess(comment) {
  return {
    type: CREATE_COMMENT_SUCCESS,
    comment,
  };
}

/**
 * Dispatched when posting the comment fails
 *
 * @return {object} - An action object with a type of CREATE_COMMENT_SUCCESS
 */
export function createCommentError(error) {
  return {
    type: CREATE_COMMENT_ERROR,
    error,
  };
}

/**
 * Like article, this action starts the request saga
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE
 */
export function likeArticle() {
  return {
    type: LIKE_ARTICLE,
  };
}

/**
 * Dispatched when posting the like is acknowledged by the request saga
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE_SUCCESS
 */
export function likeArticleSuccess() {
  return {
    type: LIKE_ARTICLE_SUCCESS,
  };
}

/**
 * Dispatched when posting the like fails
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE_ERROR
 */
export function likeArticleError(error) {
  return {
    type: LIKE_ARTICLE_ERROR,
    error,
  };
}

/**
 * Load article, this action starts the request saga
 *
 * @param  {object}  query  - The comment data
 *
 * @return {object}         - An action object with a type of LOAD_ARTICLE
 */
export function loadArticle(query) {
  return {
    type: LOAD_ARTICLE,
    query,
  };
}

/**
 * Dispatched when the article is loaded by the request saga
 *
 * @param  {object}  article  - The article data
 *
 * @return {object}           - An action object with a type of LOAD_ARTICLE_SUCCESS
 */
export function loadArticleSuccess(article) {
  return {
    type: LOAD_ARTICLE_SUCCESS,
    article,
  };
}

/**
 * Dispatched when loading the article fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_ARTICLE_ERROR
 */
export function loadArticleError(error) {
  return {
    type: LOAD_ARTICLE_ERROR,
    error,
  };
}

/**
 * Load comments, this action starts the request saga
 *
 * @return {object} - An action object with a type of LOAD_COMMENTS
 */
export function loadComments() {
  return {
    type: LOAD_COMMENTS,
  };
}

/**
 * Dispatched when the comments are loaded by the request saga
 *
 * @param  {array}  comments  - The comments data
 *
 * @return {object}           - An action object with a type of LOAD_COMMENT_SUCCESS
 */
export function loadCommentsSuccess(comments) {
  return {
    type: LOAD_COMMENTS_SUCCESS,
    comments,
  };
}

/**
 * Dispatched when loading the comments fails
 *
 * @param  {object} error - The error
 *
 * @return {object}       - An action object with a type of LOAD_COMMENTS_ERROR
 */
export function loadCommentsError(error) {
  return {
    type: LOAD_COMMENTS_ERROR,
    error,
  };
}
