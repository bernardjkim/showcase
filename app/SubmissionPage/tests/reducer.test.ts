import submissionPageReducer, { initialState } from '../reducer';

import { clearState, submitForm, submitFormError, submitFormSuccess } from '../actions';
import { SubmissionState } from '../types';

const form = {
  title: 'title',
  uri: 'uri',
  github: 'github',
  description: 'description',
  tags: [],
  screenshot: undefined,
};
describe('submissionPageReducer', () => {
  let state: SubmissionState;
  beforeEach(() => {
    state = initialState;
  });

  it('handle the clear state action', () => {
    expect(submissionPageReducer(state, clearState())).toMatchSnapshot();
  });

  it('handle the submit form action', () => {
    expect(submissionPageReducer(state, submitForm(form))).toMatchSnapshot();
  });

  it('handle the submit form success action', () => {
    expect(submissionPageReducer(state, submitFormSuccess())).toMatchSnapshot();
  });

  it('handle the submit form error action', () => {
    expect(submissionPageReducer(state, submitFormError({}))).toMatchSnapshot();
  });
});
