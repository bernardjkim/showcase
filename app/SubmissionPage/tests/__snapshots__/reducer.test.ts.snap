// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submissionPageReducer handle the clear state action 1`] = `
Object {
  "error": false,
  "loading": false,
  "submissionSuccess": false,
}
`;

exports[`submissionPageReducer handle the submit form action 1`] = `
Object {
  "error": false,
  "loading": true,
  "submissionSuccess": false,
}
`;

exports[`submissionPageReducer handle the submit form error action 1`] = `
Object {
  "error": Object {},
  "loading": false,
  "submissionSuccess": false,
}
`;

exports[`submissionPageReducer handle the submit form success action 1`] = `
Object {
  "error": false,
  "loading": false,
  "submissionSuccess": true,
}
`;
